src/twisted/plugins/twisted_reactors.py

```diff
@@ -21,42 +21,78 @@
 kqueue = Reactor(
     'kqueue', 'twisted.internet.kqreactor', 'kqueue(2)-based reactor.')
 
-cf = Reactor(
-    'cf' , 'twisted.internet.cfreactor',
-    'CoreFoundation integration reactor.')
-
 __all__ = [
-    "default", "select", "poll", "epoll", "kqueue", "cf",
+    "default", "select", "poll", "epoll", "kqueue",
 ]
 
-if _PY3:
+if requireModule("twisted.internet.cfreactor") is not None:
+    cf = Reactor(
+        'cf' , 'twisted.internet.cfreactor',
+        'CoreFoundation integration reactor.')
+
+    __all__.extend([
+         "cf"
+    ])
+
+
+if requireModule("twisted.internet.asyncioreactor") is not None:
```

Comment:
You could factor out the repeated `requireModule` calls into a helper that operates on `Reactor` instances, then filter a complete list of `Reactor` instances through it and only put the names of the ones that remain in `__all__`.  Then, the helper could also have a test.